packages/common-all/src/dnode.ts

```diff
@@ -94,7 +94,8 @@ export class DNodeUtils {
   }
 
   static basename(nodePath: string, rmExtension?: boolean) {
-    if (rmExtension) {
+    //skip for nodePath that has a sub-hierarchy starting with .md eg: learn.mdone.test, learn.md-one.new
+    if (rmExtension && nodePath.endsWith(".md")) {
```

Comment:
minor nitpick: could simplify this the rest of this part, if path ends with ".md" then `nodePath = nodePath.slice(undefined, -3)`

Reply:
Updated. Thanks!